src/TaskEither.ts

```diff
@@ -174,6 +183,50 @@ export const fromPredicate = <L, A>(predicate: Predicate<A>, whenFalse: (a: A) =
   return a => fromEither(f(a))
 }
 
+/**
+ * @function
+ * @since 1.9.0
+ */
+export const getSemigroup = <L, A>(S: Semigroup<A>): Semigroup<TaskEither<L, A>> => {
+  const S2 = taskGetSemigroup(eitherGetSemigroup<L, A>(S))
+  return {
+    concat: (x, y) => new TaskEither<L, A>(S2.concat(x.value, y.value))
+  }
+}
+
+/**
+ * @function
+ * @since 1.9.0
+ */
+export const getMonoid = <L, A>(M: Monoid<A>): Monoid<TaskEither<L, A>> => {
```

Comment:
This is unlawful, note that there's no `getMonoid` in the `Either` module.

Counterexample

```ts
it('right unit and a left', () => {
  return M.concat(left(delay(10, 'a')), M.empty)
    .run()
    .then(x => assert.deepEqual(x, eitherLeft('a'))) // fails
})
```

Reply:
Was already wondering. I will remove the `getMonoid` function.